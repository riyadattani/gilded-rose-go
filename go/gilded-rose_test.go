package main

import (
	"testing"

	approvals "github.com/approvals/go-approval-tests"
)

func Test_Foo(t *testing.T) {
	names := []string{"Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"}
	sellins := []int{0, 1, 10, 11, 5, 6, -1}
	quality := []int{0, 1, 49, 50, -1}

	approvals.VerifyAllCombinationsFor3(t, "update quality", func(name, sellin, quality interface{}) string {
		item := &Item{name.(string), sellin.(int), quality.(int)}
		items := Items{item}
		UpdateQuality(items)
		return items.String()
	}, names, sellins, quality)
}
