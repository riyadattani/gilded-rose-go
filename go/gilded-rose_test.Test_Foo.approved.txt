update quality


[Aged Brie,0,0] => &item.Item{Name:"Aged Brie", sellIn:-1, quality:2}

[Aged Brie,0,1] => &item.Item{Name:"Aged Brie", sellIn:-1, quality:3}

[Aged Brie,0,49] => &item.Item{Name:"Aged Brie", sellIn:-1, quality:50}

[Aged Brie,0,50] => &item.Item{Name:"Aged Brie", sellIn:-1, quality:50}

[Aged Brie,0,-1] => &item.Item{Name:"Aged Brie", sellIn:-1, quality:1}

[Aged Brie,1,0] => &item.Item{Name:"Aged Brie", sellIn:0, quality:1}

[Aged Brie,1,1] => &item.Item{Name:"Aged Brie", sellIn:0, quality:2}

[Aged Brie,1,49] => &item.Item{Name:"Aged Brie", sellIn:0, quality:50}

[Aged Brie,1,50] => &item.Item{Name:"Aged Brie", sellIn:0, quality:50}

[Aged Brie,1,-1] => &item.Item{Name:"Aged Brie", sellIn:0, quality:0}

[Aged Brie,10,0] => &item.Item{Name:"Aged Brie", sellIn:9, quality:1}

[Aged Brie,10,1] => &item.Item{Name:"Aged Brie", sellIn:9, quality:2}

[Aged Brie,10,49] => &item.Item{Name:"Aged Brie", sellIn:9, quality:50}

[Aged Brie,10,50] => &item.Item{Name:"Aged Brie", sellIn:9, quality:50}

[Aged Brie,10,-1] => &item.Item{Name:"Aged Brie", sellIn:9, quality:0}

[Aged Brie,11,0] => &item.Item{Name:"Aged Brie", sellIn:10, quality:1}

[Aged Brie,11,1] => &item.Item{Name:"Aged Brie", sellIn:10, quality:2}

[Aged Brie,11,49] => &item.Item{Name:"Aged Brie", sellIn:10, quality:50}

[Aged Brie,11,50] => &item.Item{Name:"Aged Brie", sellIn:10, quality:50}

[Aged Brie,11,-1] => &item.Item{Name:"Aged Brie", sellIn:10, quality:0}

[Aged Brie,5,0] => &item.Item{Name:"Aged Brie", sellIn:4, quality:1}

[Aged Brie,5,1] => &item.Item{Name:"Aged Brie", sellIn:4, quality:2}

[Aged Brie,5,49] => &item.Item{Name:"Aged Brie", sellIn:4, quality:50}

[Aged Brie,5,50] => &item.Item{Name:"Aged Brie", sellIn:4, quality:50}

[Aged Brie,5,-1] => &item.Item{Name:"Aged Brie", sellIn:4, quality:0}

[Aged Brie,6,0] => &item.Item{Name:"Aged Brie", sellIn:5, quality:1}

[Aged Brie,6,1] => &item.Item{Name:"Aged Brie", sellIn:5, quality:2}

[Aged Brie,6,49] => &item.Item{Name:"Aged Brie", sellIn:5, quality:50}

[Aged Brie,6,50] => &item.Item{Name:"Aged Brie", sellIn:5, quality:50}

[Aged Brie,6,-1] => &item.Item{Name:"Aged Brie", sellIn:5, quality:0}

[Aged Brie,-1,0] => &item.Item{Name:"Aged Brie", sellIn:-2, quality:2}

[Aged Brie,-1,1] => &item.Item{Name:"Aged Brie", sellIn:-2, quality:3}

[Aged Brie,-1,49] => &item.Item{Name:"Aged Brie", sellIn:-2, quality:50}

[Aged Brie,-1,50] => &item.Item{Name:"Aged Brie", sellIn:-2, quality:50}

[Aged Brie,-1,-1] => &item.Item{Name:"Aged Brie", sellIn:-2, quality:1}

[Backstage passes to a TAFKAL80ETC concert,0,0] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:-1, quality:0}

[Backstage passes to a TAFKAL80ETC concert,0,1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:-1, quality:0}

[Backstage passes to a TAFKAL80ETC concert,0,49] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:-1, quality:0}

[Backstage passes to a TAFKAL80ETC concert,0,50] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:-1, quality:0}

[Backstage passes to a TAFKAL80ETC concert,0,-1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:-1, quality:0}

[Backstage passes to a TAFKAL80ETC concert,1,0] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:0, quality:3}

[Backstage passes to a TAFKAL80ETC concert,1,1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:0, quality:4}

[Backstage passes to a TAFKAL80ETC concert,1,49] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:0, quality:50}

[Backstage passes to a TAFKAL80ETC concert,1,50] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:0, quality:50}

[Backstage passes to a TAFKAL80ETC concert,1,-1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:0, quality:2}

[Backstage passes to a TAFKAL80ETC concert,10,0] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:9, quality:2}

[Backstage passes to a TAFKAL80ETC concert,10,1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:9, quality:3}

[Backstage passes to a TAFKAL80ETC concert,10,49] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:9, quality:50}

[Backstage passes to a TAFKAL80ETC concert,10,50] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:9, quality:50}

[Backstage passes to a TAFKAL80ETC concert,10,-1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:9, quality:1}

[Backstage passes to a TAFKAL80ETC concert,11,0] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:10, quality:1}

[Backstage passes to a TAFKAL80ETC concert,11,1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:10, quality:2}

[Backstage passes to a TAFKAL80ETC concert,11,49] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:10, quality:50}

[Backstage passes to a TAFKAL80ETC concert,11,50] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:10, quality:50}

[Backstage passes to a TAFKAL80ETC concert,11,-1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:10, quality:0}

[Backstage passes to a TAFKAL80ETC concert,5,0] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:4, quality:3}

[Backstage passes to a TAFKAL80ETC concert,5,1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:4, quality:4}

[Backstage passes to a TAFKAL80ETC concert,5,49] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:4, quality:50}

[Backstage passes to a TAFKAL80ETC concert,5,50] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:4, quality:50}

[Backstage passes to a TAFKAL80ETC concert,5,-1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:4, quality:2}

[Backstage passes to a TAFKAL80ETC concert,6,0] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:5, quality:2}

[Backstage passes to a TAFKAL80ETC concert,6,1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:5, quality:3}

[Backstage passes to a TAFKAL80ETC concert,6,49] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:5, quality:50}

[Backstage passes to a TAFKAL80ETC concert,6,50] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:5, quality:50}

[Backstage passes to a TAFKAL80ETC concert,6,-1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:5, quality:1}

[Backstage passes to a TAFKAL80ETC concert,-1,0] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:-2, quality:0}

[Backstage passes to a TAFKAL80ETC concert,-1,1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:-2, quality:0}

[Backstage passes to a TAFKAL80ETC concert,-1,49] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:-2, quality:0}

[Backstage passes to a TAFKAL80ETC concert,-1,50] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:-2, quality:0}

[Backstage passes to a TAFKAL80ETC concert,-1,-1] => &item.Item{Name:"Backstage passes to a TAFKAL80ETC concert", sellIn:-2, quality:0}

[Sulfuras, Hand of Ragnaros,0,0] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:0, quality:0}

[Sulfuras, Hand of Ragnaros,0,1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:0, quality:1}

[Sulfuras, Hand of Ragnaros,0,49] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:0, quality:49}

[Sulfuras, Hand of Ragnaros,0,50] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:0, quality:50}

[Sulfuras, Hand of Ragnaros,0,-1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:0, quality:-1}

[Sulfuras, Hand of Ragnaros,1,0] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:1, quality:0}

[Sulfuras, Hand of Ragnaros,1,1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:1, quality:1}

[Sulfuras, Hand of Ragnaros,1,49] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:1, quality:49}

[Sulfuras, Hand of Ragnaros,1,50] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:1, quality:50}

[Sulfuras, Hand of Ragnaros,1,-1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:1, quality:-1}

[Sulfuras, Hand of Ragnaros,10,0] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:10, quality:0}

[Sulfuras, Hand of Ragnaros,10,1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:10, quality:1}

[Sulfuras, Hand of Ragnaros,10,49] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:10, quality:49}

[Sulfuras, Hand of Ragnaros,10,50] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:10, quality:50}

[Sulfuras, Hand of Ragnaros,10,-1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:10, quality:-1}

[Sulfuras, Hand of Ragnaros,11,0] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:11, quality:0}

[Sulfuras, Hand of Ragnaros,11,1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:11, quality:1}

[Sulfuras, Hand of Ragnaros,11,49] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:11, quality:49}

[Sulfuras, Hand of Ragnaros,11,50] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:11, quality:50}

[Sulfuras, Hand of Ragnaros,11,-1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:11, quality:-1}

[Sulfuras, Hand of Ragnaros,5,0] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:5, quality:0}

[Sulfuras, Hand of Ragnaros,5,1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:5, quality:1}

[Sulfuras, Hand of Ragnaros,5,49] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:5, quality:49}

[Sulfuras, Hand of Ragnaros,5,50] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:5, quality:50}

[Sulfuras, Hand of Ragnaros,5,-1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:5, quality:-1}

[Sulfuras, Hand of Ragnaros,6,0] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:6, quality:0}

[Sulfuras, Hand of Ragnaros,6,1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:6, quality:1}

[Sulfuras, Hand of Ragnaros,6,49] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:6, quality:49}

[Sulfuras, Hand of Ragnaros,6,50] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:6, quality:50}

[Sulfuras, Hand of Ragnaros,6,-1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:6, quality:-1}

[Sulfuras, Hand of Ragnaros,-1,0] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:-1, quality:0}

[Sulfuras, Hand of Ragnaros,-1,1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:-1, quality:1}

[Sulfuras, Hand of Ragnaros,-1,49] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:-1, quality:49}

[Sulfuras, Hand of Ragnaros,-1,50] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:-1, quality:50}

[Sulfuras, Hand of Ragnaros,-1,-1] => &item.Item{Name:"Sulfuras, Hand of Ragnaros", sellIn:-1, quality:-1}

[Conjured,0,0] => &item.Item{Name:"Conjured", sellIn:-1, quality:0}

[Conjured,0,1] => &item.Item{Name:"Conjured", sellIn:-1, quality:0}

[Conjured,0,49] => &item.Item{Name:"Conjured", sellIn:-1, quality:47}

[Conjured,0,50] => &item.Item{Name:"Conjured", sellIn:-1, quality:48}

[Conjured,0,-1] => &item.Item{Name:"Conjured", sellIn:-1, quality:-1}

[Conjured,1,0] => &item.Item{Name:"Conjured", sellIn:0, quality:0}

[Conjured,1,1] => &item.Item{Name:"Conjured", sellIn:0, quality:0}

[Conjured,1,49] => &item.Item{Name:"Conjured", sellIn:0, quality:48}

[Conjured,1,50] => &item.Item{Name:"Conjured", sellIn:0, quality:49}

[Conjured,1,-1] => &item.Item{Name:"Conjured", sellIn:0, quality:-1}

[Conjured,10,0] => &item.Item{Name:"Conjured", sellIn:9, quality:0}

[Conjured,10,1] => &item.Item{Name:"Conjured", sellIn:9, quality:0}

[Conjured,10,49] => &item.Item{Name:"Conjured", sellIn:9, quality:48}

[Conjured,10,50] => &item.Item{Name:"Conjured", sellIn:9, quality:49}

[Conjured,10,-1] => &item.Item{Name:"Conjured", sellIn:9, quality:-1}

[Conjured,11,0] => &item.Item{Name:"Conjured", sellIn:10, quality:0}

[Conjured,11,1] => &item.Item{Name:"Conjured", sellIn:10, quality:0}

[Conjured,11,49] => &item.Item{Name:"Conjured", sellIn:10, quality:48}

[Conjured,11,50] => &item.Item{Name:"Conjured", sellIn:10, quality:49}

[Conjured,11,-1] => &item.Item{Name:"Conjured", sellIn:10, quality:-1}

[Conjured,5,0] => &item.Item{Name:"Conjured", sellIn:4, quality:0}

[Conjured,5,1] => &item.Item{Name:"Conjured", sellIn:4, quality:0}

[Conjured,5,49] => &item.Item{Name:"Conjured", sellIn:4, quality:48}

[Conjured,5,50] => &item.Item{Name:"Conjured", sellIn:4, quality:49}

[Conjured,5,-1] => &item.Item{Name:"Conjured", sellIn:4, quality:-1}

[Conjured,6,0] => &item.Item{Name:"Conjured", sellIn:5, quality:0}

[Conjured,6,1] => &item.Item{Name:"Conjured", sellIn:5, quality:0}

[Conjured,6,49] => &item.Item{Name:"Conjured", sellIn:5, quality:48}

[Conjured,6,50] => &item.Item{Name:"Conjured", sellIn:5, quality:49}

[Conjured,6,-1] => &item.Item{Name:"Conjured", sellIn:5, quality:-1}

[Conjured,-1,0] => &item.Item{Name:"Conjured", sellIn:-2, quality:0}

[Conjured,-1,1] => &item.Item{Name:"Conjured", sellIn:-2, quality:0}

[Conjured,-1,49] => &item.Item{Name:"Conjured", sellIn:-2, quality:47}

[Conjured,-1,50] => &item.Item{Name:"Conjured", sellIn:-2, quality:48}

[Conjured,-1,-1] => &item.Item{Name:"Conjured", sellIn:-2, quality:-1}
